apps/beeswax/src/beeswax/models.py

```diff
@@ -471,7 +471,17 @@ class Session(models.Model):
   objects = SessionManager()
 
   def get_handle(self):
-    secret, guid = HiveServerQueryHandle.get_decoded(secret=self.secret, guid=self.guid)
+    secret = self.secret
+    guid = self.guid
+    if sys.version_info[0] > 2:
+      # only for py3, after bytes saved, bytes wrapped in a string object
+      # a new guid/secret before saved, the eval() will throws exception
+      try:
+        secret = eval(secret)
```

Comment:
Did you check the type of secret and even have a few values that way easy to see when it errors/not error? (can reuse them in the test)

Reply:
Added type checking and reuse in unit test